### DISCUSSION TITLE
v3.3.2

### DISCUSSION THREAD

**Discussion by danielroe** [nuxt/nuxt#19910] [Announcements]
> **3.3.2** is a patch release with plenty of bug fixes.

## 👉 Changelog

[compare changes](https://github.com/nuxt/nuxt/compare/v3.3.1...v3.3.2)

### 🔥 Performance

  - **nuxt:** Experimentally disable vue server renderer nitro endpoint ([#19825](https://github.com/nuxt/nuxt/pull/19825))

### 🩹 Fixes

  - **kit:** Provide name to `performance.mark()` ([#19687](https://github.com/nuxt/nuxt/pull/19687))
  - **nuxt:** Unpause DOM updates on suspense resolve ([#19740](https://github.com/nuxt/nuxt/pull/19740))
  - **kit:** Handle node 14 performance behaviour ([#19733](https://github.com/nuxt/nuxt/pull/19733))
  - **webpack:** Transpile app directory ([#19773](https://github.com/nuxt/nuxt/pull/19773))
  - **nuxt:** Unset context after app is created ([#19753](https://github.com/nuxt/nuxt/pull/19753))
  - **cli:** Watch dist and register restart hook after nuxt is ready ([#19736](https://github.com/nuxt/nuxt/pull/19736))
  - **nuxt:** Ignore falsy modules ([#19684](https://github.com/nuxt/nuxt/pull/19684))
  - **nuxt:** Use `h3` utilities to set response status/code ([#19713](https://github.com/nuxt/nuxt/pull/19713))
  - **nuxt:** Add temporary augmentation for webstorm (and docs) ([#19400](https://github.com/nuxt/nuxt/pull/19400))
  - **nuxt:** Handle external navigation to api routes ([#19829](https://github.com/nuxt/nuxt/pull/19829))
  - **nuxt:** Observe slot element in custom nuxt-link ([#19802](https://github.com/nuxt/nuxt/pull/19802))
  - **nuxt:** Directly render server components ([#19605](https://github.com/nuxt/nuxt/pull/19605))
  - **vite:** Support multiple rollup entries ([#19842](https://github.com/nuxt/nuxt/pull/19842))
  - **nuxt:** Ignore schema types that eval to any ([#19835](https://github.com/nuxt/nuxt/pull/19835))
  - **nuxt:** Use prerender cache for islands ([#19822](https://github.com/nuxt/nuxt/pull/19822))
  - **nuxt:** Add missing import in islands template ([#19870](https://github.com/nuxt/nuxt/pull/19870))
  - **kit:** Check if nuxt is restarting before updating templates ([#19830](https://github.com/nuxt/nuxt/pull/19830))
  - **test-utils:** Allow overriding nitro options ([#19872](https://github.com/nuxt/nuxt/pull/19872))
  - **kit:** Add legacy entrypoints for pre v3.3 usage ([#19874](https://github.com/nuxt/nuxt/pull/19874))

### 📖 Documentation

  - Fix auto-imports example ([#19520](https://github.com/nuxt/nuxt/pull/19520), [#19690](https://github.com/nuxt/nuxt/pull/19690))
  - Update seo and meta page ([#19697](https://github.com/nuxt/nuxt/pull/19697))
  - **introduction:** First batch of improvements ([98b9afa6d](https://github.com/nuxt/nuxt/commit/98b9afa6d))
  - Update module author guide ([#18551](https://github.com/nuxt/nuxt/pull/18551))
  - Add 'going further' section for custom routing ([#19565](https://github.com/nuxt/nuxt/pull/19565))
  - Improve installation prerequisites ([c570ae272](https://github.com/nuxt/nuxt/commit/c570ae272))
  - Update custom routing link ([8d644903b](https://github.com/nuxt/nuxt/commit/8d644903b))
  - Add backticks ([#19721](https://github.com/nuxt/nuxt/pull/19721))
  - Add link to layout transitions ([#19770](https://github.com/nuxt/nuxt/pull/19770))
  - Fix typo ([#19775](https://github.com/nuxt/nuxt/pull/19775))
  - Fix typo ([#19779](https://github.com/nuxt/nuxt/pull/19779))
  - Fix typo ([#19785](https://github.com/nuxt/nuxt/pull/19785))
  - Add missing comma ([#19791](https://github.com/nuxt/nuxt/pull/19791))
  - Update legacy server examples ([#19797](https://github.com/nuxt/nuxt/pull/19797))
  - Third-party state management and migration guide ([#19798](https://github.com/nuxt/nuxt/pull/19798))
  - Link to module request issue template ([#19754](https://github.com/nuxt/nuxt/pull/19754))
  - Improve swr documentation ([#18743](https://github.com/nuxt/nuxt/pull/18743))
  - Add another example for `useAsyncData` ([#19225](https://github.com/nuxt/nuxt/pull/19225))
  - Avoid `$fetch` in top-level `<script setup>` ([#19357](https://github.com/nuxt/nuxt/pull/19357))
  - Removing deprecated coming soon banner ([#19817](https://github.com/nuxt/nuxt/pull/19817))
  - Add serialization section to data fetching ([#19336](https://github.com/nuxt/nuxt/pull/19336))
  - Add protocol to superjson link ([cebfcb3da](https://github.com/nuxt/nuxt/commit/cebfcb3da))
  - Update play on X links ([8c11498a5](https://github.com/nuxt/nuxt/commit/8c11498a5))
  - Add missing `return` statement ([fc7867fb0](https://github.com/nuxt/nuxt/commit/fc7867fb0))
  - Replace `@nuxt/kit` example with node built-ins ([#19873](https://github.com/nuxt/nuxt/pull/19873))
  - Add layers video by LearnVue ([d3d09d5d3](https://github.com/nuxt/nuxt/commit/d3d09d5d3))
  - Improve components page ([340725bb0](https://github.com/nuxt/nuxt/commit/340725bb0))

### 🏡 Chore

  - Add internal types for nitro/app ([#19745](https://github.com/nuxt/nuxt/pull/19745))
  - Remove unused TS directive ([0b5ba6c3e](https://github.com/nuxt/nuxt/commit/0b5ba6c3e))
  - Bump package versions ([6d7161270](https://github.com/nuxt/nuxt/commit/6d7161270))
  - Bump package versions ([2580cc830](https://github.com/nuxt/nuxt/commit/2580cc830))
  - Bump package versions ([3ee97b125](https://github.com/nuxt/nuxt/commit/3ee97b125))

### ✅ Tests

  - Remove sourcemaps from minimal test ([#19730](https://github.com/nuxt/nuxt/pull/19730))
  - Bump bundle size ([e57c797f2](https://github.com/nuxt/nuxt/commit/e57c797f2))
  - Alias test-utils back to src again ([8b06b06a0](https://github.com/nuxt/nuxt/commit/8b06b06a0))

### 🤖 CI

  - Test built packages ([#19767](https://github.com/nuxt/nuxt/pull/19767))

<hr /><em>This discussion was created from the release <a href='https://github.com/nuxt/nuxt/releases/tag/v3.3.2'>v3.3.2</a>.</em>

**Comment by lkul2912** [comment]
Hi. My Name's Long Kul. Yêu thử thách của mọi người và vẫn chờ mọi người hỗ trợ. Tôi thật sự nghèo lắm mà.!. Kul.apple291292@gmail.com
0948291292
Hãy nhắn tin cho tôi khi hiệp hội cần nha.
Yêu Inc, MetaY và chính phủ của các bạn .!.
